test/extended/builds/cluster_config.go

```diff
@@ -0,0 +1,119 @@
+	var (
+		buildFixture           = exutil.FixturePath("testdata", "builds", "test-build.yaml")
+		defaultConfigFixture   = exutil.FixturePath("testdata", "builds", "cluster-config.yaml")
+		blacklistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-blacklist.yaml")
+		searchConfigFixture    = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-search.yaml")
+		whitelistConfigFixture = exutil.FixturePath("testdata", "builds", "cluster-config", "registry-whitelist.yaml")
+		oc                     = exutil.NewCLI("build-cluster-config", exutil.KubeConfigPath())
+	)
+
+	g.Context("", func() {
+
+		g.BeforeEach(func() {
+			exutil.PreTestDump()
+		})
+
+		g.JustBeforeEach(func() {
+			g.By("waiting for default service account")
+			err := exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "default")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("waiting for builder service account")
+			err = exutil.WaitForServiceAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()), "builder")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.Run("create").Args("-f", buildFixture).Execute()
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+				exutil.DumpConfigMapStates(oc)
+			}
+			oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+		})
+
+		g.Context("registries config context", func() {
+
+			g.It("should default registry search to docker.io for image pulls", func() {
+				g.By("apply default cluster configuration")
+				err := oc.AsAdmin().Run("apply").Args("-f", defaultConfigFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 1s for build controller configuration to propagate")
+				time.Sleep(1 * time.Second)
+				g.By("starting build sample-build and waiting for success")
+				br, err := exutil.StartBuildAndWait(oc, "sample-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertSuccess()
+				g.By("expecting the build logs to indicate docker.io is the search registry")
+				buildLog, err := br.LogsNoTimestamp()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildLog).To(o.ContainSubstring("defaulting registry to docker.io"))
+			})
+
+			g.It("can override registry search for image pulls", func() {
+				g.Skip("TEMP: initial builder image pull does not seem to be respecting the registries.conf override")
+				g.By("apply image search cluster configuration")
+				err := oc.AsAdmin().Run("apply").Args("-f", searchConfigFixture).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 1s for build controller configuration to propagate")
+				time.Sleep(1 * time.Second)
+				g.By("starting build sample-build and waiting for success")
+				br, err := exutil.StartBuildAndWait(oc, "sample-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertSuccess()
```

Comment:
shouldn't you expect this to fail?  since the build uses "centos/ruby-22-centos7" with a search list of "quay.io", and quay.io/centos/ruby-22-centos7 doesn't exist?

Reply:
didn't realize we are not mirroring our scl/s2i builder images on quay (yet?).

Reply:
dunno if there's any plan to or not.  Anyway failure is preferred because it'll fail faster than it'll succeed.  In fact i'd suggest you make this test use a non-existent registry as the search target.  That will also keep it from getting broken in the future if the image did get mirrored to quay.